kernel/src/syscall/rt_sigreturn.rs

```diff
@@ -41,11 +44,16 @@ pub fn sys_rt_sigreturn(ctx: &Context, user_ctx: &mut UserContext) -> Result<Sys
     } else {
         thread_local.sig_context().set(Some(ucontext.uc_link));
     };
-    ucontext
-        .uc_mcontext
-        .inner
-        .gp_regs
-        .copy_to_raw(user_ctx.general_regs_mut());
+    ucontext.uc_mcontext.copy_user_regs_to(user_ctx);
+
+    // Restore FPU context on stack
+    let mut fpu_context = FpuContext::new();
+    let mut fpu_context_writer = VmWriter::from(fpu_context.as_bytes_mut());
+    ctx.user_space().read_bytes(
+        ucontext.uc_mcontext.fpu_context_addr(),
```

Comment:
In RISC-V, I'm afraid that `fpu_context_addr` returns a kernel address, which does not work well with `read_bytes`.
```rust
    pub fn fpu_context_addr(&self) -> Vaddr {
        &self.fpu_context as *const _ as Vaddr
    }
```